power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -36,12 +36,12 @@ inline void get_component_sequence(MainModelState<ComponentContainer> const& sta
                                    ForwardIterator end, OutputIterator destination, Idx n_comp_elements) {
     using UpdateType = typename Component::UpdateType;
 
-    if (n_comp_elements == na_Idx) {
+    if (n_comp_elements < 0) {
```

Comment:
this was an actual bug. it received `-1` instead of `na_Idx`. While I fixed the usage, I considered making it explicit like this better.

now that I think about it, we should rename the `get_n_elements()` function to more explicitly explain that it's ONLY in case of `qualify_for_optional_id()`. The logic is currently correct, but not at first sight.

Reply:
I don't think making the naming of this one more explicit is needed. It is very short and the `return` statement explicitly shows that this is intended for `qualify_for_optional_id()`. But we can briefly bring it up. I'll leave this one open to see if anyone has a saying about it.

Reply:
It should actually be `invalid_index` as other places. My typo I think.

Reply:
i think it's good as is. `<0` captures both the `na_Idx` and `invalid_index` edge cases. Do you agree with the current proposal?